sections/06_discussion.md

```diff
@@ -49,6 +49,86 @@ with only a couple GPUs.*
+illustrative. See [@tag:Stein2010_cloud] for example of calculating storage
+cost,
+and [@tag:NIH2016_genome_cost] for data on the cost of sequencing.*
+
+Many have sought to curb the costs of deep learning, with methods ranging from
+the very applied (e.g., reduced numerical precision [@tag:Gupta2015_prec
+@tag:Bengio015_prec @tag:Sa2015_buckwild @tag:Hubara2016_qnn]) to the exotic
+and theoretic (e.g., training small networks to mimic large networks and
+ensembles [@tag:Caruana2014_need @tag:Hinton2015_dark_knowledge]). The largest
+gains in efficiency have come from computation with graphics processing units
+(GPUs) [@tag:Raina2009_gpu @tag:Vanhoucke2011_cpu @tag:Seide2014_parallel
+@tag:Hadjas2015_cc @tag:Edwards2015_growing_pains
+@tag:Schmidhuber2014_dnn_overview], which excel at the matrix and vector
+operations so central to deep learning. The massively parallel nature of GPUs
+allows additional optimizations, such as accelerated mini-batch gradient
+descent [@tag:Vanhoucke2011_cpu @tag:Seide2014_parallel @tag:Su2015_gpu
+@tag:Li2014_minibatch]. However, GPUs also have a limited quantity of memory,
+which makes it difficult to train/use networks of significant size and
+complexity on a single GPU or machine [@tag:Raina2009_gpu
+@tag:Krizhevsky2013_nips_cnn]. This restriction has sometimes stymied the use
+of deep learning for computational biology[@tag:Chen2016_gene_expr], though
+some have chosen to use slower CPU implementations rather than sacrifice
+network size or performance [@tag:Yasushi2016_cgbvs_dnn].
+
+Steady improvements in GPU hardware may alleviate this issue somewhat, but it
+is not clear whether it can occur quickly enough to keep up with the increasing
+amount of available biological data. Much has been done to minimize the memory
+requirements of neural networks [@tag:CudNN @tag:Caruana2014_need
+@tag:Gupta2015_prec @tag:Bengio015_prec @tag:Sa2015_buckwild
+@tag:Chen2015_hashing @tag:Hubara2016_qnn], but there is also growing
+interest in specialized hardware, such as field-programmable gate arrays
+(FPGAs) [@tag:Edwards2015_growing_pains Lacey2016_dl_fpga] and
+application-specific integrated circuits (ASICs). Specialized hardware promises
+improvements in deep learning at reduced time, energy, and memory
+[@tag:Edwards2015_growing_pains]. Logically, there is less software for highly
+specialized hardware [@tag:Lacey2016_dl_fpga], and it could be a difficult
+investment for those not solely interested in deep learning. However, it is
+likely that such options will find increased support as they become a more
+popular platform for deep learning and general computation.
+
+Distributed computing is a general solution to intense computational
+requirements, and has enabled many large-scale deep learning efforts. Early
+approaches to distributed computation were not suitable for deep learning
+[@tag:Dean2012_nips_downpour], but significant progress has been made. There
```

Comment:
Was citing the discussion provided in @tag:Dean2012_nips_downpour, rather than Downpour itself. Updated with clarification.